dao/mongo.go

```diff
@@ -431,24 +431,55 @@ func (m *MongoService) GetAttachmentFromInsolvencyResource(transactionID string,
 	if err != nil {
 		if err == mongo.ErrNoDocuments {
 			log.Debug("no insolvency case found for transaction id", log.Data{"transaction_id": transactionID})
-			return nil, nil
+			return models.AttachmentResourceDao{}, nil
 		}
 
 		log.Error(err)
-		return nil, err
+		return models.AttachmentResourceDao{}, err
 	}
 
 	err = storedAttachment.Decode(&insolvencyResource)
 	if err != nil {
 		log.Error(err)
-		return nil, err
+		return models.AttachmentResourceDao{}, err
 	}
 
-	// Return an empty array instead of nil to distinguish from insolvency case
-	// not found
-	if insolvencyResource.Data.Attachments == nil {
-		return make([]models.AttachmentResourceDao, 0), nil
+	return insolvencyResource.Data.Attachments[0], nil
+}
+
+func (m *MongoService) DeleteAttachmentResource(transactionID, attachmentID string) (int, error) {
+	collection := m.db.Collection(m.CollectionName)
+
+	// Choose specific transaction for insolvency case with attachment to be removed
+	filter := bson.M{"transaction_id": transactionID}
+
+	// Check if insolvency case exists for specified transactionID
+	storedInsolvency := collection.FindOne(context.Background(), filter)
+	err := storedInsolvency.Err()
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			log.Error(err)
+			return http.StatusNotFound, fmt.Errorf("there was a problem handling your request for transaction id [%s] - insolvency case not found", transactionID)
```

Comment:
Shouldn't return an error here. http.StatusNotFound is enough to indicate that no document was found

Reply:
The error is used in the message response for the user to distinguish from other error messages in that method. It's also consistent with what we've done in other parts of our app